src/renderer/shared/lib/utils/arrays.ts

```diff
@@ -8,3 +10,29 @@
 export function splice<T extends any>(collection: T[], item: T, position: number): T[] {
   return [...collection.slice(0, position), item, ...collection.slice(position + 1)];
 }
+
+/**
+ * Create dictionary with given key and value
+ * Keys can only be type of string, number or symbol
+ * @param collection array of items
+ * @param property field to be used as key
+ * @param predicate transformer function
+ * @return {Object}
+ */
+export function dictionary<T extends Record<K, PropertyKey>, K extends KeysOfType<T, PropertyKey>>(
```

Comment:
Same as `keyBy` from _lodash_ except it has TS support for valid keys (`string | number | symbol` only)
and provide optional predicate function if we require some transformed value